File: Produto-Service/src/main/java/com/microchefs/produtoservice/dto/DLQSupportDTO.java
package com.microchefs.produtoservice.dto;

import java.time.LocalDateTime;

public record DLQSupportDTO(
        String tipoMensagem,
        String filaDeOrigem,
        String tipoErro,
        String mensagemDeErro,
        String mensagemOriginal,
        LocalDateTime timestamp
) {}

File: Usuario-Service/DTOs/Events/DLQSupportDTO.cs
using System.Text.Json.Serialization;

namespace ClienteService.DTOs.Events
{
    public record DLQSupportDTO
    (
        [property: JsonPropertyName("tipoMensagem")] string TipoMensagem,
        [property: JsonPropertyName("filaDeOrigem")] string FilaDeOrigem,
        [property: JsonPropertyName("tipoErro")] string TipoErro,
        [property: JsonPropertyName("mensagemDeErro")] string MensagemDeErro,
        [property: JsonPropertyName("mensagemOriginal")] string MensagemOriginal,
        [property: JsonPropertyName("timestamp")] DateTime TimeStamp
    );
}
